game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -1342,6 +1356,53 @@ private static void placeUnits(final TriggerAttachment t, final Territory terr,
     bridge.addChange(change);
   }
 
+  /**
+   * Test if we are resetting/clearing the variable first, and if so, remove the leading "-reset-" or "-clear-"
+   * from the new value.
+   */
+  public static Tuple<Boolean, String> getClearFirstNewValue(final String preNewValue) {
+    final boolean clearFirst = preNewValue.length() > 0
+        && (preNewValue.startsWith(PREFIX_CLEAR) || preNewValue.startsWith(PREFIX_RESET));
+    final String newValue = clearFirst
+        ? preNewValue.replaceFirst(PREFIX_CLEAR, "").replaceFirst(PREFIX_RESET, "")
+        : preNewValue;
+    return Tuple.of(clearFirst, newValue);
+  }
+
+  /**
+   * Returns the property change as well as history start event message.
+   *
+   * <p>
+   * No side-effects.
+   * </p>
+   */
+  static Optional<Tuple<Change, String>> getPropertyChangeHistoryStartEvent(
+      final TriggerAttachment triggerAttachment,
+      final DefaultAttachment propertyAttachment, final String propertyName,
+      final Tuple<Boolean, String> clearFirstNewValue,
+      final String propertyAttachmentName, final Named attachedTo) {
+    final boolean clearFirst = clearFirstNewValue.getFirst();
+    final String newValue = clearFirstNewValue.getSecond();
+
+    final boolean isValueTheSame = newValue.equals(propertyAttachment.getRawPropertyString(propertyName));
+
+    if (!isValueTheSame) {
+
+      final Change change = clearFirst && newValue.isEmpty()
+          ? ChangeFactory.attachmentPropertyReset(propertyAttachment, propertyName)
+          : ChangeFactory.attachmentPropertyChange(propertyAttachment, newValue, propertyName, clearFirst);
+
+      final String startEvent =
+          MyFormatter.attachmentNameToText(triggerAttachment.getName()) + ": Setting " + propertyName
+              + (newValue.length() > 0 ? " to " + newValue : " cleared ") + " for "
+              + propertyAttachmentName + " attached to " + attachedTo.getName();
```

Comment:
I believe `String#format` would definitely help making this thing a lot more readable:
```java
final String startEvent = String.format("%s: Setting %s %s for %s attached to %s",
    MyFormatter.attachmentNameToText(triggerAttachment.getName()),
    propertyName,
    !newValue.isEmpty() ? "to " + newValue : "cleared",
    propertyAttachmentName,
    attachedTo.getName());
```

Note how this helped me uncover some issues with duplicate spaces (haven't actually noticed them before writing this code).